imager/ImagerProfile/models.py

```diff
@@ -30,40 +31,44 @@ class ImagerProfile(models.Model):
     objects = models.Manager()
     active = ActiveImagerManager()
 
-    def __str__(self):
-        return "User: {}".format(self.user.username)
-
-    def is_active(self):
-        return self.user.is_active()
-
-    following = models.ManyToManyField('self', symmetrical=False)
-
-    def follow(self, other_profile):
-        return self.following.add(other_profile.ImagerProfile)
-
-    def unfollow(self, other_profile):
-        return self.following.remove(other_profile.ImagerProfile)
-
-    def following(self):
+    following = models.ManyToManyField('self',
+                                       symmetrical=False,
+                                       null=True,
+                                       blank=True,
+                                       related_name='followers')
+    blocking = models.ManyToManyField('self',
+                                      symmetrical=False,
+                                      null=True,
+                                      blank=True,
+                                      related_name='+')
+
+    def follow(self, imagerprofile):
+        if self in imagerprofile.blocking.all():
+            return 'User has blocked you'
+        if self is imagerprofile:
+            return 'You cannot follow yourself'
```

Comment:
For this, and line 47 above, I'd suggest raising an error instead of returning a string.  It'll be more Pythonic to do so and then catch the error in calling code than to do this.